js/runner.go

```diff
@@ -540,6 +542,10 @@ type VU struct {
 	setupData goja.Value
 
 	state *lib.State
+	// ID of this VU in each scenario
+	scenarioID map[string]uint64
+	// count of iterations executed by this VU in each scenario
+	scenarioIter map[string]uint64
```

Comment:
Are these maps just because of the `ramping-vus` executor?

Reply:
Yeah, I thought it would be simpler to just keep the count for all scenarios in cases where VUs are released and reused, though it would be more efficient to just use `uint64` here instead. Mihail made the same comment above. I'll see if I can change it.

Reply:
Actually, taking a look at this again, I'm not sure if changing it would be better. The memory needed for these maps is negligible (a few entries at most?), but we need some state to know if a VU ID was previously assigned by that same executor:
  https://github.com/k6io/k6/blob/f9e0ab6550c21477dbdfcb6cd49be8d755a47f11/js/runner.go#L620-L622

This is what drives the `BaseExecutor.VUIDLocal` counter (which could probably be unexported), as otherwise `GetNextScVUID()` would have to be called from elsewhere. Maybe in each executor after `GetPlannedVU()`:
  https://github.com/k6io/k6/blob/f9e0ab6550c21477dbdfcb6cd49be8d755a47f11/lib/executor/shared_iterations.go#L277

But that would mean several implementations of it, when using a map is simpler and we wouldn't need to think about this for new executors that might reuse VUs.

Let me know if there's another reason both you and @MStoykov dislike this approach and well, any suggestions. :)